scipy/sparse/_index.py

```diff
@@ -266,8 +266,13 @@ def _unpack_index(index):
         else:
             raise IndexError('invalid number of indices')
     else:
-        row, col = index, slice(None)
-
+        idx = _compatible_boolean_index(index)
+        if idx is None:
+            row, col = index, slice(None)
+        elif idx.ndim < 2:
+            return _boolean_index_to_array(idx), slice(None)
+        elif idx.ndim == 2:
```

Comment:
Earlier `col` was set to `None` for non `numpy.ndarray`s  irrespective of the dimension which is why the 2D index array couldn't have worked.